core/src/main/java/io/grpc/util/RoundRobinLoadBalancerFactory.java

```diff
@@ -368,14 +369,17 @@ synchronized Subchannel getSubchannel(String stickinessValue) {
     @Nullable
     private final RoundRobinLoadBalancer.StickinessState stickinessState;
     @SuppressWarnings("unused")
-    private volatile int index = -1; // start off at -1 so the address on first use is 0.
+    private volatile int index;
 
-    Picker(
-        List<Subchannel> list, @Nullable Status status,
-        @Nullable RoundRobinLoadBalancer.StickinessState stickinessState) {
+    Picker(List<Subchannel> list, @Nullable Status status,
+           @Nullable RoundRobinLoadBalancer.StickinessState stickinessState) {
       this.list = list;
       this.status = status;
       this.stickinessState = stickinessState;
+      // start off with a random address to ensure significant Picker churn does not skew
+      // subchannel selection lower-indexed addresses in the list
+      Random random = new Random();
```

Comment:
Don't re-create the Random instance so frequently. Instead, have it a field of RoundRobinLoadBalancer and probably pass the value (or chosen starting index) into the Picker constructor. Since the Pickers are only created from one thread at a time, there's no concurrency issues from using Random at that point.